data/domain/cards.tsx

```diff
@@ -60,6 +65,17 @@ export class Card {
         }
     }
 
+    getBonusID = (): number => {
```

Comment:
Can you add some comments, I'm not fully grasping what this does.

Is this simply looking for the matching bonus in the map and fetching the id of it? 

I also haven't look at the rest of the code but if we call this more then once, I think we should persist this into an attribute after we find it once, it won't change so feels wasteful to do it multiple times. (again maybe you don't, so ignore me if you don't)

Reply:
Indeed storing the bonusID is a clever idea, would be dumb to calculate it again and again as it won't change